src/game.cpp

```diff
@@ -3368,7 +3368,7 @@ bool game::save_achievements()
             return !std::isgraph( c, locale );
         }, '_' );
     } else {
-        achievement_file_path << u.name;
+        achievement_file_path << cata::filter_invalid_filename_char( u.name );
```

Comment:
This can simply reuse the code from line 3365-3369 above to replace special characters with `_` in _both_ if branches instead of adding a new function.

Reply:
ok, I'll look into this, last time I saw code above, but it's actually not do anything, this time I'll try to use a forloop, then transfer the other PR into a draft, if this works successfully.

Reply:
Accomplish! :D